zegami_sdk/source.py

```diff
@@ -335,3 +336,49 @@ def _get_mime_type(cls, path) -> str:
         if ext in cls.IMAGE_MIMES.keys():
             return cls.IMAGE_MIMES[ext]
         raise TypeError('"{}" is not a supported image mime_type ({})'.format(path, cls.IMAGE_MIMES))
+
+
+class UrlSource(UploadableSource):
+
+    def __init__(self, name, url_template, image_fetch_headers, column_filename='Filename'):
+        """Used in conjunction with create_collection().
+
+        A UrlSource() fetches the images from the url template given, resulting in the
+        generation of a true Source() in the collection.
+        """
+        self.name = name
+        self.url_template = url_template
+        self.image_fetch_headers = image_fetch_headers
+        self.column_filename = column_filename
+
+        # Set externally once a blank collection has been made
+        self._source = None
+        self._index = None
+    
+    def _upload(self):
```

Comment:
I suppose '_upload' is a bit of an inaccurate name here, but I think that's fine.
I think this is fine, but a purer approach would be to have a common base class that both this and UploadableSource inherit from, with a method called something like 'apply' or 'execute'. Probably more trouble than it's worth though.

Reply:
Worth a print statement to mirror the one for UploadableSource, e.g. 'configuring source to fetch images from urls'.